### PR TITLE
Do more complete error checking in get_pool_blockdevs

### PR DESCRIPTION
**Pull request by mulkieran**

### CODE REVIEW

**Comment by mulkieran** [comment]
@agrover @trgill review?

File: src/engine/strat_engine/setup.rs [hunk 1]
```diff
@@ -217,7 +217,17 @@ pub fn get_pool_blockdevs(devnode_table: &HashMap<PoolUuid, Vec<PathBuf>>,
...
         for dev in devnodes {
+            let device = try!(Device::from_str(&dev.to_string_lossy()));
+
+            // If we've seen this device already, skip it.
+            if devices.contains(&device) {
+                continue;
+            } else {
+                devices.insert(device);
+            }
+
```
Comment (trgill) [inline src/engine/strat_engine/setup.rs]: What is the case this detects?

Comment (mulkieran) [inline src/engine/strat_engine/setup.rs]: Two device nodes that refer to the same device. Unusual, but possible.

**Comment by agrover** [comment]
last reviewer, 👍, merging

**Pull request merged**